salt/modules/glusterfs.py

```diff
@@ -117,11 +133,20 @@ def list_peers():
 
     '''
     root = _gluster_xml('peer status')
-    result = [x.find('hostname').text for x in _iter(root, 'peer')]
-    if len(result) == 0:
+    if not _gluster_ok(root):
         return None
-    else:
-        return result
+
+    result = []
+    for peer in _iter(root, 'peer'):
+        hostname = peer.find('hostname').text
+        hostnames = peer.find('hostnames')
+        if hostnames is not None:
+            for alternative in hostnames:
+                if alternative.text.startswith(hostname + '.'):
+                    hostname = alternative.text
+        result.append(hostname)
+
+    return result
```

Comment:
I thought about that, and chose a dict with the hostname as the key for the aliases because any code that used the list of hostnames **as** a list will still work. An iterator is still going to walk through the keys, which will be the list of hostnames.